discovery/discover.js

```diff
@@ -0,0 +1,55 @@
+
+
+const BASE_URL = "https://ion.tbd.engineering";
+const DID_TYPE_ENDPOINT = "/didtype/1669";
+const IDENTIFIER_PREFIX = "/identifiers/";
+
+async function fetchDIDList() {
+  const response = await fetch(BASE_URL + DID_TYPE_ENDPOINT);
+  if (!response.ok) {
+    throw new Error('Failed to fetch DID list');
+  }
+  return await response.json();
+}
+
+async function fetchDIDData(did) {
+  console.log(BASE_URL + IDENTIFIER_PREFIX + did);
+  const response = await fetch(BASE_URL + IDENTIFIER_PREFIX + did);
+  if (!response.ok) {
+    throw new Error('Failed to fetch DID data for ' + did);
+  }
+  return await response.json();
+}
+
+async function fetchPFIServiceEndpoints() {
+  const ids = await fetchDIDList();
+  const promises = ids.map(id => {
+    const ionDid = "did:ion:" + id;
+    return fetchDIDData(ionDid);
+  });
+
+  const didDataList = await Promise.all(promises);
+
+  const pfiServiceEndpoints = didDataList.reduce((results, didData) => {
+    const services = didData.didDocument.service;
+    const pfiServices = services.filter(service => service.type === 'PFI');
+    
+    if (pfiServices.length > 0) {
+      results.push({
+        did: didData.didDocument.id,
+        serviceEndpoint: pfiServices[0].serviceEndpoint
+      });
+    }
+
+    return results;
+  }, []);
+
+  return pfiServiceEndpoints;
+}
+
+// Exporting the function to be used in other parts of your app
+export default fetchPFIServiceEndpoints;
+
+
+const endpoints = await fetchPFIServiceEndpoints();
+console.log(endpoints);
```

Comment:
this bit did not work for me when i ran node discover.js - seemed to complain about the export bit and the await bit. 

changing this seemed to work though. maybe not the best way to do it but i was able to get the output i was looking for.

```javascript
exports.fetchPFIServiceEndpoints = fetchPFIServiceEndpoints;

async function main() {
  const endpoints = await fetchPFIServiceEndpoints();
  console.log(endpoints)
}

main().then(() => process.exit(0), e => { console.error(e); process.exit(1) })```

Reply:
oh. seems like changing the extension to `discover.mjs` worked :) same for create-and-anchor.js

Reply:
yes - needs to have a package.json with type module - I should mention that.